### DISCUSSION TITLE
Create new volume from video stream.

### DISCUSSION THREAD

**Discussion by rskelly** [biigle/core#845] [Ideas]
We're wondering about the feasibility of using a hosted video stream as a volume source. For example, a stream from [Ocean Networks Canada](https://data.oceannetworks.ca/7291c862-3e96-4c50-9443-7c6d494259b9) (link to actual stream; we have not consulted with ONC about the use of streams for this purpose, it's just an example). 

Currently, one must enter the URL of a directory containing video files and list the files individually. Hypothetically, one could enter the stream URL without the list of individual files and stream the volume from that location.

This might be tangentially related to issue #273.

**Comment by mzur** [comment]
This is pretty much what we originally wanted in https://github.com/biigle/core/issues/273. However, we now dropped the idea of attempting to display the live video in BIIGLE (for various reasons). Instead, we like the idea of a separate "live annotation" application (on a laptop, tablet or phone) that you can use to create annotations while watching the live stream. This could be useful on a ship with a live ROV video and ship telemetry. Or it could be used with something like the ONC stream. After the stream finishes, the captured video and the live annotations could be uploaded to BIIGLE as files, and merged with some sort of timestamp synchronization. In the end you will have a (preliminarily) annotated video in BIIGLE.

**Reply by rskelly** [reply]
Thanks for the feedback Martin. 

I guess the easiest way to accomplish this with existing toolset is just to download the stream using a plugin and use it as a static file.

**Comment by merlinbest5** [comment]
to me these are 2 separate tasks. the live annotation would be useful, but we have other tools that can do it at the moment. for this issue i'm hoping to plant the idea that being able to link to a stream (not live) would be very cool. i often hear from groups that struggle with server storage but would like to annotate with biigle online, and if all of this beautiful video is already hosted freely on seatube then downloading it and uploading it somewhere else seems like wasted space and effort. i have no understanding of the mechanisms involved, just putting my vote in :)

for anyone interested, here's a link to an example video stream on seatube: https://data.oceannetworks.ca/SeaTubeV3?resourceTypeId=600&resourceId=6820&time=2023-09-12T20:06:08.843Z&annotationId=11510310

**Reply by mzur** [reply]
Now I see! This is indeed a different use case than the live annotation. Unfortunately, the "mechanisms involved" for the SeaTube stream are quite different to what BIIGLE requires for processing. So there will be no easy solution to integrate this. But I'd be open for a chat with the SeaTube developers to check out the options. Would you like to arrange that? I'd prefer email (unless the discussion gets too complicated) because of the time zone difference 😉

**Reply by merlinbest5** [reply]
happy to arrange something! stay tuned.
